assets/src/js/media-frame.js

```diff
@@ -7,14 +7,55 @@
  */
 import React from 'react';
 import { createRoot } from '@wordpress/element';
+import { Snackbar } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
-import { observeElement, getFrameTitle, getFrameProperty, isBrandSite } from './utils';
+import { observeElement, getFrameTitle, getFrameProperty, isBrandSite, getNoticeClass, showSnackbarNotice } from './utils';
 import BrowserUploaderButton from '../admin/media-sharing/browser-uploader';
-import { isSyncAttachment as isSyncAttachmentApi } from '../components/api';
+
+/**
+ * Get sync status from attachment model/element.
+ * The data is already available from the admin-ajax.php response.
+ *
+ * @param {HTMLElement|Object} attachmentOrElement - Attachment model or DOM element.
+ *
+ * @return {boolean} Whether the attachment is synced.
+ */
+function isSyncAttachment( attachmentOrElement ) {
+	// If it's a DOM element with data attributes
+	if ( attachmentOrElement instanceof HTMLElement ) {
+		const attachmentId = attachmentOrElement.dataset.id || attachmentOrElement.dataset.attachmentId;
+
+		if ( ! attachmentId ) {
+			return false;
+		}
+
+		// Try to get from Backbone model
+		const wpMedia = getFrameProperty( 'wp.media' );
+		if ( wpMedia && wpMedia.attachment ) {
+			const attachment = wpMedia.attachment( attachmentId );
+			if ( attachment && attachment.get ) {
+				const syncStatus = attachment.get( 'is_sync_attachment' );
+				// Check if the value exists and is truthy
+				return syncStatus === true || syncStatus === 1 || syncStatus === '1';
+			}
+		}
+
+		// Fallback: check if element has the sync class (set by showSyncBadge)
+		return attachmentOrElement.classList.contains( 'onemedia-synced-media' );
```

Comment:
Very much beyond the scope, but I promise you'll be a much happier developer if you get into the habit of returning early until only the happy path is left, instead of sandwiching the happy path between failed paths and edge cases.